### PR TITLE
Update distribution support information

### LINKED ISSUE #1108: Improve documentation regarding supported distributions
## Problem description

> [ ERROR   ]: 14:20:24 | KiwiBootStrapPhaseFailed: Bootstrap package installation failed: Unable to detect release version (use '--releasever' to specify release version)

## Expected behaviour
no error?
Config can be found here:
[https://gist.github.com/ybaumy/f557e36b28e74473e16f2ad1dac002dd](config.xml)
## OS and Software information
+ KIWI version: 
> KIWI (next generation) version 9.17.39
+ Operating system: 
SLES15SP0
+ OBS version:

### PR DESCRIPTION
**Pull request by schaefi**
Update the documentation with regards to our supported
distribution statement. This Fixes #1108

### CODE REVIEW

**Comment by schaefi** [comment]
@tomschr would be great if you can have a look. I'm not super happy with my wording but also couldn't come up with something better... just too hot in Germany.

Thanks

File: doc/source/building.rst [hunk 1]
```diff
@@ -61,16 +61,91 @@ Building Images
...
+   +------------------+------------+---------------+------------------+----------+-------------+-----------+
+   | Ubuntu 16        | no         | no            | no               | no       | no          | yes       |
+   +------------------+------------+---------------+------------------+----------+-------------+-----------+
+
+.. admonition:: dnf
+
+   dnf is the package manager used on Fedora and RHEL and is
+   the successor of yum. When kiwi builds images for this distributions
+   the latest version of dnf is required to be installed on the host to
+   build the image
```
Comment (tomschr) [inline doc/source/building.rst]:
```suggestion
   build the image.
```

File: doc/source/building.rst [hunk 2]
```diff
@@ -61,16 +61,91 @@ Building Images
...
+
+In general our goal is to support any major distribution with KIWI. However
+for building images we rely on core tools which are not under our control.
+Also several design aspects of distributions are different and not
+influenced by us. There are many side effects that can be annoying especially
+if the building host is not of the same distribution vendor than the image
+target.
+
+Supported Platforms/Architectures
```
Comment (tomschr) [inline doc/source/building.rst]: Hmn,... :thinking: ... maybe better use "and" for "/":

```suggestion
Supported Platforms and Architectures
```

File: doc/source/building.rst [hunk 3]
```diff
@@ -61,16 +61,91 @@ Building Images
...
+.. admonition:: dnf
+
+   dnf is the package manager used on Fedora and RHEL and is
+   the successor of yum. When kiwi builds images for this distributions
+   the latest version of dnf is required to be installed on the host to
+   build the image
+
+In general our goal is to support any major distribution with KIWI. However
+for building images we rely on core tools which are not under our control.
+Also several design aspects of distributions are different and not
```
[the remark this answers is not in the document]
Comment (schaefi) [reply doc/source/building.rst]: I made that more clear

File: doc/source/building.rst [hunk 4]
```diff
@@ -61,16 +61,91 @@ Building Images
...
-
-For anything older please consider to use the legacy KIWI version *v7.x*
-For more details on the legacy kiwi, see: :ref:`legacy_kiwi`.
+KIWI can build images for the distributions which are **equal** or **newer**
+compared to the table below. For anything older please consider to use the
+legacy KIWI version *v7.x* For more details on the legacy kiwi,
+see: :ref:`legacy_kiwi`
+
+The most compatible environment is provided if the build host is of the same
+distribution than the target image. For images build with the
```
[the remark this answers is not in the document]
Comment (schaefi) [reply doc/source/building.rst]: I'll use the shorter version :)

**Comment by schaefi** [comment]
Thanks much @tomschr :) I updated the docs, should be good now

**Pull request merged**